### DISCUSSION TITLE
Release 2.5.2 broke support for PHP 7.4

### DISCUSSION THREAD

**Discussion by ghost** [spatie/icalendar-generator#105] [Bugs] [answered]
The trailing comma for function arguments, introduced in [1], isn't supported in PHP 7.4, resulting in a ParseError:
```
ParseError: syntax error, unexpected ')', expecting variable (T_VARIABLE)

vendor/spatie/icalendar-generator/src/Components/Event.php:355
```

That's why the pipeline also fails for 7.4 [2]. I also wonder how a failing pipeline allows for a release?

[1] https://github.com/spatie/icalendar-generator/pull/104/files#diff-12ca5b67c9f5fef1d4b4134cff800aa328f9af9cb98fa2fbdc851da74ca8cfefR14
[2] https://github.com/spatie/icalendar-generator/actions/runs/3970764656/jobs/6830759965

**Comment by VaguelyOnline** [comment]
We have found this exact issue also. Needing to rollback to earlier version to account for this.

**Comment by freekmurze** [comment]
I've tagged a new release that fixes the syntax error on PHP 7.4 

There are a couple of more tests failing which I don't have time to fix them now (I would appreciate a PR)

**Reply by VaguelyOnline** [reply]
Thanks for looking into this and responding so quickly.

**Comment by ghost** [comment, marked as answer]
Unfortunately 2.5.3 fixes only the one occurrence. I probably should have linked them all. I'm looking into providing a PR.

EDIT: https://github.com/spatie/icalendar-generator/pull/106
EDIT: PR got merged and 2.5.4 released

**Reply by ghost** [reply]
Ok, just updated to the 2.5.4 and our downsteam tests work again. Thanks for the fast release.
